src/common/settings.ts

```diff
@@ -127,6 +149,19 @@ function getGlobalValue<T>(config: WorkspaceConfiguration, key: string, defaultV
   return inspect?.globalValue ?? inspect?.defaultValue ?? defaultValue;
 }
 
+function getOptionalGlobalValue<T>(config: WorkspaceConfiguration, key: string): T | undefined {
+  const inspect = config.inspect<T>(key);
+  return inspect?.globalValue;
+}
+
+function getGlobalLineLength(config: WorkspaceConfiguration, key: string): number | undefined {
+  let lineLength = config.inspect<number>(key)?.globalValue;
+  if (lineLength && lineLength > 0 && lineLength <= 320) {
+    return lineLength;
+  }
+  return undefined;
+}
```

Comment:
Nit. We could avoid the repetion by passing the line-length value instead

```suggestion
function toLineLength(lineLength: number | undefined): number | undefined {
  if (lineLength && lineLength > 0 && lineLength <= 320) {
    return lineLength;
  }
  return undefined;
}
```

It can then be used in combination with `getOptionalGlobalValue`